bdikit/utils.py

```diff
@@ -70,3 +73,26 @@ def get_gdc_layered_metadata():
             metadata[key] = (subschema, data)
 
     return metadata
+
+
+def hash_dataframe(df: pd.DataFrame) -> str:
```

Comment:
In `bdikit/visualization.py`, Eden has implemented this using a pandas function `pd.util.hash_pandas_object`. Could we use the same approach he used here as well?

Reply:
I am unsure if that approach is collision-safe regarding column names. As this is minor, I am leaving this as it is. We can revisit this function at any time.

Reply:
Sounds good to me. I think @EdenWuyifan should look at this anyway and see if the caching in `bdikit/visualization.py` is still necessary after we merge PR. We certainly don't want to cache GDC embeddings twice.

Reply:
I am using this line since pandas provide this function which will hash any DataFrame/Series: ```hashlib.sha1(pd.util.hash_pandas_object(self.source).values).hexdigest()```

Reply:
```pd.util.hash_pandas_object(self.source)``` is sensitive to both column names and values. If we only need to focus on column names, we can use this new hash function instead.